Source/CarthageKit/Dependency.swift

```diff
@@ -4,14 +4,23 @@ import Tentacle
 
 /// Uniquely identifies a project that can be used as a dependency.
 public enum Dependency {
+
+	public struct Binary {
+		// Actual Absolute URL
+		let absoluteURL: URL
```

Comment:
Potential for better names options? This is the value resolved to load the resource

Reply:
What do you think of:

```swift
protocol URLConvertible: CustomStringConvertible {
    var url: URL {get}
}

struct BinaryURL: URLConvertible {
    
    let url: URL
    let overloadedDescription: String?
    var description: String {
        
        return overloadedDescription ?? url.description
    }
    
}


BinaryURL(url: URL.init(string: "https://google.com")!, overloadedDescription: "mom!")
```

Reply:
Interesting. I like the Convertible naming friendliness with Swift convertible protocol.

Will experiment with this approach.

Reply:
Does not look like to me that the developer using the code nor the code structure requires it to be distinct. That's my opinion however. A few more eyes on this would be nice.